6.x/crud-filters.md

```diff
@@ -5,63 +5,176 @@
 <a name="about"></a>
 ## About
 
-Backpack CRUD allows you to show a filters bar right above the entries table. When selected or modified, they reload the DataTables results. The search will then search within the filtered elements.
+Backpack allows you to show a filters bar right above the entries table. When selected or modified, they reload the DataTable. The search bar will also take filters into account, only looking within filtered results.
 
 ![Backpack CRUD Filters](https://backpackforlaravel.com/uploads/docs-4-0/filters/filters.png)
 
-Just like with fields, columns or buttons, you can add existing filters or create a custom filter that fits to your particular needs. Everything's done inside your ```EntityCrudController::setupListOperation()```. 
+Just like with fields, columns or buttons, you can add existing filters or create a custom filter that fits to your particular needs. Everything's done inside your ```EntityCrudController::setupListOperation()```.
 
-This is a <span class="badge badge-info">PRO</span> feature. It requires that you have [purchased access to `backpack/pro`](https://backpackforlaravel.com/products/pro-for-unlimited-projects).
+> **Note:** This is a <span class="badge badge-info">PRO</span> feature. It requires that you have [purchased access to `backpack/pro`](https://backpackforlaravel.com/pricing).
 
 <a name="methods"></a>
 ### Filters API
 
 In order to manipulate filters, you can use:
 
 ```php
-$this->crud->addFilter($options, $values, $filter_logic);
+// on one filter
+CRUD::filter($name)
+  ->type($type)
+  ->whenActive($closure)
+  ->whenInactive($closure)
+  ->apply();
+
+CRUD::filter($name)->remove();
+CRUD::filter($name)->makeFirst();
+CRUD::filter($name)->makeLast();
+CRUD::filter($name)->before($different_filter_name);
+CRUD::filter($name)->after($different_filter_name);
+
+// on all filters
+CRUD::removeAllFilters(); // removes all the filters
+CRUD::filters(); // gets all the filters
+```
 
-$this->crud->removeFilter($name);
-$this->crud->removeAllFilters();
+<a name="adding-a-filter"></a>
+### Adding and configuring a filter
 
-$this->crud->filters(); // gets all the filters
+Inside your `setupListOperation()` you can add or select a filter using `CRUD::filter('name')`, then chain methods to completely configure it:
 
-// aditionally you should be able to change Filter attributes by appending `->methods` in your filter definition. Eg - this would set the filter attribute `type` to be equal to `number`:
-$this->crud->addFilter($options, $values, $filter_logic)->type('number');
+```php
+CRUD::filter('name')
+    ->type('text')
+    ->label('The name')
+    ->whenActive(function($value) {
+        CRUD::addClause('where', 'name', 'LIKE', '%'.$value.'%');
+    })->else(function($value) {
```

Comment:
@tabacitu else closure causing an error `Too few arguments to function {closure}`

Reply:
Oh that's odd... what's the solution here @karandatwani92

Reply:
This works perfectly without $value `->else(function() { }` and it's right.

My test says:
->withActive() //$value is passed
->else() //$value not passed

->whenInActive() //$value not passed
->else() //$value is `null` passed.

Reply:
Ok thank you

```suggestion
    })->else(function() {
```